action/src/plugins/imageDiff/plugin.ts

```diff
@@ -0,0 +1,80 @@
+import * as core from '@actions/core';
+import { randomUUID } from 'crypto';
+import fs from 'fs';
+import download from 'image-downloader';
+import { basename } from 'path';
+import pixelmatch from 'pixelmatch';
+import { PNG } from 'pngjs';
+
+import { ImageDiffPlugin } from '../../schemas/stoatConfigSchema';
+import { GithubActionRun, UploadImageDiffRequest } from '../../types';
+import { submitPartialConfig } from '../helpers';
+import { createSignedUrl, uploadPath } from '../staticHosting/helpers';
+
+const runImageDiffPlugin = async (
+  taskId: string,
+  taskConfig: ImageDiffPlugin,
+  { ghToken, ghRepository: { repo, owner }, ghSha }: GithubActionRun,
+  stoatConfigFileId: number
+) => {
+  core.info(`[${taskId}] Running static hosting plugin (stoat config ${stoatConfigFileId})`);
+  const currentDirectory = process.cwd();
+  core.info(`[${taskId}] Current directory: ${currentDirectory}`);
+
+  // get baseline image
+  if (!taskConfig.baseline) {
+    throw new Error(`[${taskId}] Missing baseline image URL in ${taskId} task config`);
+  }
+  const uuid = randomUUID();
+  const baselinePath = `${currentDirectory}/${uuid}-baseline.png`;
+  core.info(`[${taskId}] Downloading baseline image to ${baselinePath}...`);
+  const baselineImageFile = await download.image({
+    url: taskConfig.baseline,
+    dest: baselinePath
+  });
+
+  // create diff image
+  const diffPath = `${currentDirectory}/${uuid}-diff.png`;
+  core.info(`[${taskId}] Creating image diff to ${diffPath}...`);
+  const baselineImage = PNG.sync.read(fs.readFileSync(baselinePath));
+  const currentImage = PNG.sync.read(fs.readFileSync(taskConfig.path));
+  const { width, height } = baselineImage;
+  const diffImage = new PNG({ width, height });
+  pixelmatch(baselineImage.data, currentImage.data, diffImage.data, width, height, { threshold: 0.1 });
```

Comment:
Not blocking this PR at all, but ideally we should have a link to a static site that takes the user to a page that lets them do similar things to the [GitHub image comparison functionality](https://github.blog/2011-03-21-behold-image-view-modes/) except for non-checked-in images.

Reply:
Agree.